docs/root/intro/version_history.rst

```diff
@@ -21,7 +21,7 @@ Features:
 - kotlin: add a way to tell Envoy Mobile to respect system proxy settings by calling an ``enableProxying(true)`` method on the engine builder. (:issue:`#2416 <2416>`)
 - kotlin: add a ``enableSkipDNSLookupForProxiedRequests(true)`` knob for controlling whether Envoy waits on DNS response in the dynamic forward proxy filter for proxied requests. (:issue:`#2602 <2602>`)
 - api: Add various methods to C++ EngineBuilder to bring it to parity with the Java and Obj-C builders. (:issue:`#2498 <2498>`)
-
+- api: Add support for String Accessors to the C++ EngineBuilder. (:issue:`#2498 <2498>`)
```

Comment:
I'm not sure I understand what the intended use of this is. Will it be more clear in a future PR or should I push for documentation clarity here?

Reply:
Good question. So String Accessors are already "a thing" in the Java and Obj-C engine builders. Rafal explained it to me as "so with string accessor you get an access to the API that you can call from within an EM core and which will you return a string that comes from the platform layer. ... I want to implement a filter in c++ that requires me having access to an  api_token. EM does not now anything about api_token but I can register a string accessor that returns it if needed"

So this PR merely adds a C++ implementation of an Accessor and adds a method to the C++ Builder to register them when creating an engine. I *think* improved docs for this feature would be great, but is probably out of scope for this PR. WDYT?

Reply:
fair enough :-P